gateway/api/v1/views.py

```diff
@@ -0,0 +1,43 @@
+"""
+Views api for V1.
+"""
+
+from rest_framework import permissions
+
+
+from api import views
+from api.models import NestedProgram, Job
+from api.permissions import IsOwner
+from . import serializers as v1_serializers
+
+
+class NestedProgramViewSet(
+    views.NestedProgramViewSet
+):  # pylint: disable=too-many-ancestors
+    """
+    Nested program view set first version. Use NestedProgramSerializer V1.
+    """
+
+    queryset = NestedProgram.objects.all()
+    serializer_class = v1_serializers.NestedProgramSerializer
+    permission_classes = [permissions.IsAuthenticated]
+
+    @staticmethod
+    def get_serializer_job_class():
+        return v1_serializers.JobSerializer
+
+    def get_serializer_class(self):
+        return v1_serializers.NestedProgramSerializer
```

Comment:
Should this be `return self.serializer_class`?

Reply:
For this particular case not. Because what I'm doing here is override the serializer. For a new version modify the `View` would be something as easy as:

```py
    # in gateway/api/v2/views.py
    def get_serializer_class(self):
        return v2_serializers.NestedProgramSerializer
```

And both versions could live at the same time.

Reply:
Changed here @akihikokuroda [81e70aa](https://github.com/Qiskit-Extensions/quantum-serverless/pull/354/commits/81e70aa64355d7e89620609a8882278f750272e9)